core/app/api/products/ids/route.ts

```diff
@@ -0,0 +1,22 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+import { getProductsByIds } from '~/client/queries/get-products';
+
+export const GET = async (request: NextRequest) => {
+  const searchParams = request.nextUrl.searchParams;
+  const query = searchParams.get('ids');
+
+  if (query == null) {
+    return NextResponse.json({ status: 'error', error: 'ids query is required' }, { status: 400 });
+  }
+
+  const id_list = query.split(',');
```

Comment:
What are we using to set the search params in the client? If we're using `nuqs` we should use the same parser here, otherwise we risk this thing breaking if we change how params are serialized